src/Driver/ChromeDriver/Downloader.php

```diff
@@ -124,12 +127,7 @@ private function downloadArchive(Driver $driver): string
 
         $response = $this->httpClient->request(
             'GET',
-            sprintf(
-                '%s/%s/%s.zip',
-                $this->getDownloadEndpoint($driver),
-                $driver->version()->toBuildString(),
-                $this->getBinaryName($driver)
-            )
+            $this->downloadUrlResolver->byDriver($driver, $this->getBinaryName($driver)),
```

Comment:
`getBinaryName()` is only used here. I'd move it to the implementation of the `DownloadUrlResolver`, so it doesn't need to be passed here. Then the interface for `DownloadUrlResolver::byDriver($driver)` also makes more sense.